src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs

```diff
@@ -272,16 +272,20 @@ private void WriteChoiceSchema(CodeWriter writer, EnumType schema)
                     }
                     writer.Line();
 
+                    int index = 1;
```

Comment:
Shouldn't this index be calculated in GetValueFieldName?

Reply:
Ahh, right. But now I will get conflict with other property:

![image](https://user-images.githubusercontent.com/45216704/89349416-00b21a80-d663-11ea-82b6-b862e6f5dc7c.png)

Reply:
So, do we want to remove `ItemValue1` enum value or add underscore ex: `private const string Item_Value1 = "Item";`?

Reply:
That's the point, other property should get 2 prepended

Reply:
So you're saying whatever it is generating [now](https://github.com/Azure/autorest.csharp/pull/859/files#diff-e917ec1838b6d293116e7324bd7a7fc8) is correct? If yes, then why do we want to calculate index inside `GetValueFieldName`?

Reply:
Could you please tell me if we have `ItemValue` enum with  "ItemValue1", "Item", "ItemValue2" members:
```
"EnumValue": {
      "type": "string",
      "enum": [
        "ItemValue1",
        "Item",
        "ItemValue2"
      ],
      "x-ms-enum": {
        "name": "ItemValue",
        "modelAsString": true
      }
    },
```

then , what would be the private property names for "ItemValue1", "Item", "ItemValue2"?

Reply:
>So, do we want to remove ItemValue1 enum value or add underscore ex: private const string Item_Value1 = "Item";?

Sorry missed this comment.

We have multiple possible conflicts here:
1. Enum item with enum name
2. Enum item with enum backing field

Where possible we should use the defined enum item name and only workaround the backing field.

So in the following sample:

```
"EnumValue": {
      "type": "string",
      "enum": [
        "ItemValue1",
        "Item",
        "ItemValue2"
      ],
      "x-ms-enum": {
        "name": "ItemValue",
        "modelAsString": true
      }
    },
```

We should have something like this.
```
 ItemValue1 - ItemValue1Value
 Item - ItemValue3 (because 1 and 2 are taken)
 ItemValue2 - ItemValue2Value
```

Reply:
Thanks for the explanation!